### PR TITLE
APPM-1138, APPM-1167

### PR DESCRIPTION
**Pull request by sajithab**

### CODE REVIEW

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/view.asset.js [hunk 1]
```diff
@@ -48,15 +48,13 @@ $('#update-store').click(function () {
...
-                showMessageModel(msg, head, pathname);
+                showMessageModel(response.message, head, pathname);
                 e.stopPropagation();
             }
 
         },
         error: function (response) {
-            var msg = "Error adding external stores.";
-            showMessageModel(msg, head, pathname);
+            showMessageModel(response.message, head, pathname);
```
Comment (ruwanta) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/view.asset.js]: What if Response does not have message?

File: modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/subscriptions.jag [hunk 2]
```diff
@@ -78,7 +79,14 @@ require('/modules/store.js').exec(function (ctx) {
 
     }
 
-
+	//Recent app rating.
+	for (var i = 0; i < recentAssets.length; i++) {
+		if (JSON.parse(social.getRating("webapp:" + recentAssets[i].id))) {
+			recentAssets[i].rating = {average: Math.floor(JSON.parse(social.getRating("webapp:" + recentAssets[i].id)).rating)};
```
Comment (ruwanta) [inline modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/subscriptions.jag]: can change to for performance improvement. Here JSON parse seems to call twice for the same purpose.
var asset = JSON.parse(social.getRating("webapp:" + recentAssets[i].id));
if(asset != null) {
ecentAssets[i].rating = asset.rating;
} ...

File: modules/jaggery-apps/app-publisher-web/themes/appm/js/view.asset.js (hunk 1 above)
Comment (sajithab) [inline modules/jaggery-apps/app-publisher-web/themes/appm/js/view.asset.js]: response.message won't be null in this point.

File: modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/subscriptions.jag (hunk 2 above)
Comment (sajithab) [inline modules/jaggery-apps/app-store-web/extensions/assets/webapp/pages/subscriptions.jag]: Fixed - https://github.com/wso2/product-app-manager/pull/299

**Pull request merged**